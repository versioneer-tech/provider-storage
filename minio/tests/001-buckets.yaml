---
apiVersion: pkg.internal/v1beta1
kind: Storage
metadata:
  name: s-joe
spec:
  bucketAccessGrants:
  - bucketName: s-joe
    grantedAt: "2025-09-29T10:05:00Z"
    grantee: jeff
    permission: ReadWrite
  bucketAccessRequests:
  - bucketName: s-jeff-shared
    reason: Need access
    requestedAt: "2025-09-29T10:00:00Z"
  buckets:
  - bucketName: s-joe
    discoverable: true
  principal: joe
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: bucket-s-joe, observe-joe-restricted-s-jeff-shared,
      policy-full-s-joe, and 2 more'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-s-joe
  labels:
    crossplane.io/composite: s-joe
  name: bucket-s-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Bucket
      metadata:
        annotations:
          storages.pkg.internal/discoverable: "true"
        name: s-joe
      spec:
        providerConfigRef:
          name: provider-minio
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-joe-restricted-s-jeff-shared
  labels:
    crossplane.io/composite: s-joe
  name: observe-joe-restricted-s-jeff-shared
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Policy
      metadata:
        name: joe.restricted.s-jeff-shared
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-full-s-joe
  labels:
    crossplane.io/composite: s-joe
  name: policy-full-s-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Policy
      metadata:
        name: joe.full.s-joe
      spec:
        forProvider:
          rawPolicy: |-
            {
              "Version": "2012-10-17",
              "Statement": [
                {
                  "Effect": "Allow",
                  "Action": [
                    "s3:*"
                  ],
                  "Resource": [
                    "arn:aws:s3:::s-joe",
                    "arn:aws:s3:::s-joe/*"
                  ]
                }
              ]
            }
        providerConfigRef:
          name: provider-minio
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-jeff-restricted-s-joe
  labels:
    crossplane.io/composite: s-joe
  name: policy-jeff-restricted-s-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Policy
      metadata:
        name: jeff.restricted.s-joe
      spec:
        forProvider:
          rawPolicy: |-
            {
              "Version": "2012-10-17",
              "Statement": [
                {
                  "Effect": "Allow",
                  "Action": [
                    "s3:GetBucketLocation",
                    "s3:ListBucket",
                    "s3:GetObject",
                    "s3:PutObject",
                    "s3:DeleteObject"
                  ],
                  "Resource": [
                    "arn:aws:s3:::s-joe",
                    "arn:aws:s3:::s-joe/*"
                  ]
                }
              ]
            }
        providerConfigRef:
          name: provider-minio
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-joe
  labels:
    crossplane.io/composite: s-joe
  name: user-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: User
      metadata:
        name: joe
      spec:
        forProvider:
          policies:
          - joe.full.s-joe
          userName: joe
        providerConfigRef:
          name: provider-minio
        writeConnectionSecretToRef:
          name: joe
          namespace: default
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
