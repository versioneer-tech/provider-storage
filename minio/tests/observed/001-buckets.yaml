apiVersion: kubernetes.crossplane.io/v1alpha1
kind: Object
metadata:
  name: observe-joe-restricted-b-jeff-shared
  annotations:
    crossplane.io/composition-resource-name: observe-joe-restricted-b-jeff-shared
status:
  atProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Policy
      metadata:
        name: joe.restricted.b-jeff-shared
  conditions:
  - type: Ready
    status: "True"