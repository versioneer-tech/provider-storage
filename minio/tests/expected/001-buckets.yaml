---
apiVersion: pkg.internal/v1beta1
kind: Storage
metadata:
  name: s-joe
spec:
  bucketAccessGrants:
  - bucketName: b-joe
    grantee: jeff
    permission: ReadWrite
  bucketAccessRequests:
  - bucketName: b-jeff-shared
    reason: Need access
  buckets:
  - bucketName: b-joe
    discoverable: true
  principal: joe
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: bucket-b-joe, observe-joe-restricted-b-jeff-shared,
      policy-full-b-joe, and 2 more'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-b-joe
  labels:
    crossplane.io/composite: s-joe
  name: bucket-b-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Bucket
      metadata:
        annotations:
          storages.pkg.internal/discoverable: "true"
        name: b-joe
      spec:
        providerConfigRef:
          name: provider-minio
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-joe-restricted-b-jeff-shared
  labels:
    crossplane.io/composite: s-joe
  name: observe-joe-restricted-b-jeff-shared
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Policy
      metadata:
        name: joe.restricted.b-jeff-shared
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-full-b-joe
  labels:
    crossplane.io/composite: s-joe
  name: policy-full-b-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Policy
      metadata:
        name: joe.full.b-joe
      spec:
        forProvider:
          rawPolicy: |-
            {
              "Version": "2012-10-17",
              "Statement": [
                {
                  "Effect": "Allow",
                  "Action": [
                    "s3:*"
                  ],
                  "Resource": [
                    "arn:aws:s3:::b-joe",
                    "arn:aws:s3:::b-joe/*"
                  ]
                }
              ]
            }
        providerConfigRef:
          name: provider-minio
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-jeff-restricted-b-joe
  labels:
    crossplane.io/composite: s-joe
  name: policy-jeff-restricted-b-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Policy
      metadata:
        name: jeff.restricted.b-joe
      spec:
        forProvider:
          rawPolicy: |-
            {
              "Version": "2012-10-17",
              "Statement": [
                {
                  "Effect": "Allow",
                  "Action": [
                    "s3:GetBucketLocation",
                    "s3:ListBucket",
                    "s3:GetObject",
                    "s3:PutObject",
                    "s3:DeleteObject"
                  ],
                  "Resource": [
                    "arn:aws:s3:::b-joe",
                    "arn:aws:s3:::b-joe/*"
                  ]
                }
              ]
            }
        providerConfigRef:
          name: provider-minio
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-joe
  labels:
    crossplane.io/composite: s-joe
  name: user-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: User
      metadata:
        name: joe
      spec:
        forProvider:
          policies:
          - joe.full.b-joe
          userName: joe
        providerConfigRef:
          name: provider-minio
        writeConnectionSecretToRef:
          name: joe
          namespace: default
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
